proposals/001-format-api.md

```diff
@@ -0,0 +1,629 @@
+11. LCP Protected PDF
+12. Audiobook (Readium)
+13. DiViNa
+14. Web Publication (Readium)
+15. EPUB
+16. CBZ
+17. ZAB (Zipped Audio Book)
+18. LPF
+18. PDF
+
+#### Audiobook (Readium)
+
+* Light:
+  * extension is `audiobook`
+  * media type is `application/audiobook+zip`
+* Heavy:
+  * it's a ZIP archive with a `manifest.json` entry, parsed as an RWPM with `metadata.@type == http://schema.org/Audiobook`
+
+#### Audiobook Manifest (Readium)
+
+* Light:
+  * media type is `application/audiobook+json`
+* Heavy:
+  * it's a JSON file, parsed as an RWPM with `metadata.@type == http://schema.org/Audiobook`
+
+#### CBZ
+
+* Light:
+  * extension is `cbz`
+  * media type is `application/vnd.comicbook+zip`, `application/x-cbz` or `application/x-cbr`
+* Heavy ([reference](https://wiki.mobileread.com/wiki/CBR_and_CBZ)):
+  * it's a ZIP archive containing only entries with the given extensions: [`acbf`](https://wiki.mobileread.com/wiki/ACBF), `gif`, `jpeg`, `jpg`,`png`, `tiff`, `tif`, `webp` or `xml`
+  * entries starting with a `.` and `Thumbs.db` are ignored
+
+#### DiViNa
+
+* Light:
+  * extension is `divina`
+  * media type is `application/divina+zip`
+* Heavy:
+  * it's a ZIP archive with a `manifest.json` entry parsed as an RWPM, with a reading order containing only bitmap images – checked using `MediaType.isBitmap` on each `Link.type`
+
+#### DiViNa Manifest
+
+* Light:
+  * media type is `application/divina+json`
+* Heavy:
+  * it's a JSON file parsed as an RWPM, with a reading order containing only bitmap images – checked using `MediaType.isBitmap` on each `Link.type`
+
+#### EPUB
+
+* Light:
+  * extension is `epub`
+  * media type is `application/epub+zip`
+* Heavy ([reference](https://www.w3.org/publishing/epub3/epub-ocf.html#sec-zip-container-mime)):
+  * it's a ZIP archive with a `mimetype` entry containing strictly `application/epub+zip`, encoded in US-ASCII
+
+#### HTML
+
+* Light:
+  * extension is `htm`, `html`, `xht` or `xhtml`
+  * media type is `text/html` or `application/xhtml+xml`, checked using `MediaType.isHTML`
+* Heavy:
+  * it's an XML document with an `<html>` root node
+
+#### OPDS 1 Feed
+
+* Light:
+  * media type is `application/atom+xml;profile=opds-catalog`
+* Heavy:
+  * it's an XML document with a `<feed>` root node with the XML namespace `http://www.w3.org/2005/Atom`
+
+#### OPDS 1 Entry
+
+* Light:
+  * media type is `application/atom+xml;type=entry;profile=opds-catalog`
+* Heavy:
+  * it's an XML document with an `<entry>` root node with the XML namespace `http://www.w3.org/2005/Atom`
+
+#### OPDS 2 Feed
+
+* Light:
+  * media type is `application/opds+json`
+* Heavy:
+  * it's a JSON file parsed as an RWPM with a `Link` with `self` rel and `application/opds+json` type
+
+#### OPDS 2 Publication
+
+* Light:
+  * media type is `application/opds-publication+json`
+* Heavy:
+  * it's a JSON file parsed as an RWPM with at least one `Link` with a rel *starting with* `http://opds-spec.org/acquisition` 
```

Comment:
@HadrienGardeur I'm actually not sure that an OPDS 2 Publication *must* have a an acquisition link. For example, for an audiobook it could be only that, right?
```json
"links": [
    {"rel": "self", "href": "http://example.org/manifest.json", "type": "application/audiobook+json"}
]
```

In which case, I guess there's no way to differentiate an OPDS 2 Publication from another RWPM, except from the context?

Reply:
An OPDS 2.0 publication must have either:

- an acquisition link
- or a `readingOrder`

Reply:
Okay, then we can keep this "heavy" step, but if we don't have the media type or an acquisition link, then the sniffing will return `WebPub` instead of `OPDS2Publication`.